src/browser/api/system.js

```diff
@@ -689,13 +687,11 @@ exports.System = {
         }, []);
 
         if (!missingRequiredScripts.length) {
-            const missingOptionalScript = '';
-            response.scripts = preloadOption.map(preload => preloadScriptsCache[preload.url] || missingOptionalScript);
+            // when load/fetch failed, mapped value will be `undefined`, which stringify translates to `null`
+            return preloadOption.map(preload => preloadScriptsCache[preload.url]);
         } else {
-            response.error = `Execution of preload scripts canceled because of missing required script(s) ${JSON.stringify(missingRequiredScripts)}`;
+            return `Execution of preload scripts canceled because of missing required script(s) ${JSON.stringify(missingRequiredScripts)}`;
```

Comment:
No, lets return an array and handle the error in another way, either a promise or something else.

Reply:
already fixed this in a later PR, but I'll copy & paste it back to here (and rerun tests).